### PR TITLE
Docs: Restructure Advanced Guide

### PR DESCRIPTION
**Pull request by fa9r**
## Describe changes

I grouped the advanced guide pages into three different sections:

Screenshot 2023-07-19 at 18 11 18

Gitbook preview: https://app.gitbook.com/o/-MCl1Hlw9oU4xibZ-ymf/s/vlf6HVLGPRknfyMwMtR2/user-guide/advanced-guide

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Other (add details above)

### CODE REVIEW

File: docs/book/toc.md [hunk 1]
```diff
@@ -18,24 +18,24 @@
...
-  * [Data versioning and artifacts management](user-guide/advanced-guide/data-versioning-and-artifacts-lineage.md)
-  * [Schedule pipeline runs](user-guide/advanced-guide/schedule-pipeline-runs.md)
-  * [Fetch metadata within steps](user-guide/advanced-guide/fetch-metadata-within-steps.md)
-  * [Use failure/success hooks](user-guide/advanced-guide/use-failure-success-hooks.md)
-  * [Handle custom data types](user-guide/advanced-guide/handle-custom-data-types.md)
-  * [Hyperparameter tuning](user-guide/advanced-guide/hyperparameter-tuning.md)
-  * [Scale compute to the cloud](user-guide/advanced-guide/scale-compute-to-the-cloud.md)
-  * [Debug and solve issues](user-guide/advanced-guide/debug-and-solve-issues.md)
-  * [Global settings of ZenML](user-guide/advanced-guide/global-settings-of-zenml.md)
+  * [Pipelining Features](user-guide/advanced-guide/pipelining-features/pipelining-features.md)
```
Comment (bcdurak) [inline docs/book/toc.md:21]:
```suggestion
  * [Pipelining features](user-guide/advanced-guide/pipelining-features/pipelining-features.md)
```

File: docs/book/toc.md [hunk 2]
```diff
@@ -18,24 +18,24 @@
...
+    * [Configure steps/pipelines](user-guide/advanced-guide/pipelining-features/configure-steps-pipelines.md)
+    * [Compose pipelines](user-guide/advanced-guide/pipelining-features/compose-pipelines.md)
+    * [Schedule pipeline runs](user-guide/advanced-guide/pipelining-features/schedule-pipeline-runs.md)
+    * [Fetch metadata within steps](user-guide/advanced-guide/pipelining-features/fetch-metadata-within-steps.md)
+    * [Use failure/success hooks](user-guide/advanced-guide/pipelining-features/use-failure-success-hooks.md)
+    * [Hyperparameter tuning](user-guide/advanced-guide/pipelining-features/hyper-parameter-tuning.md)
+  * [Artifact Management](user-guide/advanced-guide/artifact-management/artifact-management.md)
```
Comment (bcdurak) [inline docs/book/toc.md:28]:
```suggestion
  * [Artifact management](user-guide/advanced-guide/artifact-management/artifact-management.md)
```

File: docs/book/toc.md [hunk 3]
```diff
@@ -18,24 +18,24 @@
...
+    * [Handle custom data types](user-guide/advanced-guide/artifact-management/handle-custom-data-types.md)
+    * [Visualize artifacts](user-guide/advanced-guide/artifact-management/visualize-artifacts.md)
+  * [Environment Management](user-guide/advanced-guide/environment-management/environment-management.md)
```
Comment (bcdurak) [inline docs/book/toc.md:31]:
```suggestion
  * [Environment management](user-guide/advanced-guide/environment-management/environment-management.md)
```

File: docs/book/user-guide/advanced-guide/artifact-management/artifact-management.md [hunk 4]
```diff
@@ -2,47 +2,47 @@
 description: Managing your data with ZenML.
 ---
 
-# Data versioning and artifacts management
+# Artifact Management
```
Comment (bcdurak) [inline docs/book/user-guide/advanced-guide/artifact-management/artifact-management.md:5]:
```suggestion
# Artifact management
```

File: docs/book/user-guide/advanced-guide/pipelining-features/pipelining-features.md
Comment (bcdurak) [inline docs/book/user-guide/advanced-guide/pipelining-features/pipelining-features.md:1]: As this is the second page in the advanced guide, I would suggest enriching this page with some content. An introduction paragraph, a simple picture, and the cards for navigation could look really nice here.

**Review by bcdurak — COMMENTED** [review]
Great work. I left a few small comments.

**Review by htahir1 — APPROVED** [review]

File: docs/book/user-guide/advanced-guide/pipelining-features/pipelining-features.md
Comment (fa9r) [reply docs/book/user-guide/advanced-guide/pipelining-features/pipelining-features.md:1]: Gitbook creates a table automatically if you leave the page empty

Screenshot 2023-07-20 at 09 26 13

Comment (fa9r) [reply docs/book/user-guide/advanced-guide/pipelining-features/pipelining-features.md:1]: IMO this even looks cleaner than the custom table we made in the advanced guide page, so I'm considering to just delete the contents of that page as well

Screenshot 2023-07-20 at 09 26 13

Comment (fa9r) [reply docs/book/user-guide/advanced-guide/pipelining-features/pipelining-features.md:1]: these many tiles are really overwhelming IMO and the headline and intro text don't really contain any info, so just empty page might actually look better here, I'll try that out

Comment (fa9r) [reply docs/book/user-guide/advanced-guide/pipelining-features/pipelining-features.md:1]: Okay, no, empty page looks really bad there 😅 

Screenshot 2023-07-20 at 09 34 32

**Pull request merged**